.github/workflows/main.yml

```diff
@@ -166,6 +169,71 @@ jobs:
         run:  |
           find . -name "pinetime-app.*" -ls
           find . -name "pinetime-mcuboot-app.*" -ls
+ 
+      #########################################################################################
+      # PR related tasks
+      - name: Get map for target branch
+        uses: actions/cache@v2
+        id: cache-map
+        if: github.event_name == 'pull_request'
+        with:
+          path: maps/${{ github.event.pull_request.base.sha }}.map
+          key: map-${{ github.event.pull_request.base.sha }}
+
+      - name: Checkout PR target files
+        uses: actions/checkout@v2
+        if: steps.cache-map.outputs.cache-hit != 'true' && github.event_name == 'pull_request'
+        with:
+          # using sha instead of ref as different points in time will have different map files
+          ref: ${{ github.event.pull_request.base.sha }}
+          submodules: true
+          clean: false
+
+      - name: CMake
+        if: steps.cache-map.outputs.cache-hit != 'true' && github.event_name == 'pull_request'
+        run:  |
+          rm -rf build
+          mkdir -p build maps
+          cd build
+          cmake -DARM_NONE_EABI_TOOLCHAIN_PATH=${{ runner.temp }}/arm-none-eabi -DNRF5_SDK_PATH=${{ runner.temp }}/nrf5_sdk -DUSE_OPENOCD=1 ../
+
+        #########################################################################################
+        # Make and Upload DFU Package
+        # pinetime-mcuboot-app.img must be flashed at address 0x8000 in the internal flash memory with OpenOCD:
+        # program image.bin 0x8000
+
+        # For Debugging Builds: Remove "make" option "-j" for clearer output. Add "--trace" to see details.
+        # For Faster Builds: Add "make" option "-j"
+
+      - name: Make pinetime-app
+        if: steps.cache-map.outputs.cache-hit != 'true' && github.event_name == 'pull_request'
+        run:  |
+          cd build
+          make -j pinetime-app
+          cp src/pinetime-app-1.0.0.map ../maps/${{ github.event.pull_request.base.sha }}.map 
+
+      - name: Compare map versions
+        run: |
+          python3 tools/compare_maps.py maps/${{ github.event.pull_request.base.sha }}.map maps/proposed.map
```

Comment:
build will fail on this script as it's not in the PR target (`develop`).. thought about copying it to $tmp but am not convinced